activation/post_supervisor_test.go

```diff
@@ -0,0 +1,92 @@
+package activation
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+func Test_PostSupervisor_StartsServiceCmd(t *testing.T) {
+	log := zaptest.NewLogger(t)
+
+	postDir := t.TempDir()
+	path, err := exec.Command("go", "env", "GOMOD").Output()
+	require.NoError(t, err)
+
+	opts := PostSupervisorConfig{
+		PostServiceCmd:  filepath.Join(filepath.Dir(string(path)), "build", "service"),
+		DataDir:         postDir,
+		NodeAddress:     "http://127.0.0.1:12345", // node isn't listening, but also not relevant for test
+		PowDifficulty:   DefaultPostConfig().PowDifficulty,
+		PostServiceMode: "light",
+	}
+
+	ps, err := NewPostSupervisor(log.Named("supervisor"), opts)
+	require.NoError(t, err)
+	require.NotNil(t, ps)
+	t.Cleanup(func() { assert.NoError(t, ps.Close()) })
+
+	time.Sleep(500 * time.Millisecond) // to ensure cmd actually started
+
+	pid := int(ps.pid.Load())
+	process, err := os.FindProcess(pid)
+	require.NoError(t, err)
+	require.NotNil(t, process)
+
+	if runtime.GOOS != "windows" {
+		require.NoError(t, process.Signal(syscall.Signal(0))) // check if process is running
+	}
+
+	require.NoError(t, ps.Close())
+
+	if runtime.GOOS != "windows" {
+		require.Error(t, process.Signal(syscall.Signal(0))) // check if process is closed
+	}
+}
+
+func Test_PostSupervisor_RestartsOnCrash(t *testing.T) {
+	log := zaptest.NewLogger(t)
+
+	postDir := t.TempDir()
+	path, err := exec.Command("go", "env", "GOMOD").Output()
+	require.NoError(t, err)
+
+	opts := PostSupervisorConfig{
+		PostServiceCmd:  filepath.Join(filepath.Dir(string(path)), "build", "service"),
+		DataDir:         postDir,
+		NodeAddress:     "http://127.0.0.1:12345", // node isn't listening, but also not relevant for test
+		PowDifficulty:   DefaultPostConfig().PowDifficulty,
+		PostServiceMode: "light",
+	}
+
+	ps, err := NewPostSupervisor(log.Named("supervisor"), opts)
+	require.NoError(t, err)
+	require.NotNil(t, ps)
+	t.Cleanup(func() { assert.NoError(t, ps.Close()) })
+
+	time.Sleep(500 * time.Millisecond) // to ensure cmd actually started
```

Comment:
How about using `Eventually()` with a bigger timeout instead? Otherwise, it will be flaky.

Reply:
What would the condition for `Eventually` be?

Reply:
Never mind, I think I found an assertion that works 🙂